### DISCUSSION TITLE
Access another extension's globalState

### DISCUSSION THREAD

**Discussion by sketchbuch** [microsoft/vscode-discussions#1169] [Extension Development QnA] [answered]
Can an extension access another extension's globalState? When reading the keys, only the keys for the extension where I call `context.globalState.keys()` are returned.

Is it possible to look at what another extension has written to globalState?

**Comment by alefragnani** [comment, marked as answer]
Hi @sketchbuch ,

I would say _I hope not_ 😆 

Based on the documentation (https://code.visualstudio.com/api/references/vscode-api#ExtensionContext), anything stored in the `ExtensionContext` should be _private to an extension_. So, I understand that not only `globalState` is not accessible, but also `workspaceState`, `storageUri`, and so on. 

I did a simple test, trying to load a `globalState` directly (using `get("theStateFromOtherExtension", defaultValue) `), using two of my extensions, and it didn't load.

If you really need this, maybe an alternative would be the _other extension_ to provide an API (https://code.visualstudio.com/api/references/vscode-api#extensions) for you.

Hope this helps

**Comment by sketchbuch** [comment]
Hi @alefragnani ,

Thanks! Yeah, I thought for security reason you maybe can't! I was thinking about making an extension to let people view and maybe edit globalstate data of any extension - like a list of extensions, you select one and then in another view you can see its state.

Oh well.... I will need to think of somthing else to do. Thanks!

**Comment by sketchbuch** [comment]
Actualy, maybe there is a way. The data is in a sqllite db:

![image](https://github.com/microsoft/vscode-discussions/assets/16445037/855a8022-3685-431b-ac49-4a4e1e3b6f03)

**Reply by alefragnani** [reply]
Yes, there is this SQLite database. I didn't comment about it because it isn't an _API way_. But depending on your needs, it should work.

This file stores a lot of VS Code data. In the early days, it wasn't stored in disk but used the Chromium area in Dev Tools.

BTW, I'm not entirely sure if you will have a single file for the entire VS Code data, or if each **Profile** stores its own. Based on the screenshot, it seems to be one for each **Profile**.
